# Capítulo 2

In [1]:
import numpy as np

## Ejercicio 1

Los siguientes cálculos se realizarán sin utilizar la función *round()* para redondear los decimales

### Parte A

In [2]:
muy_poco = 386
suficiente= 103
demasiado= 8
sin_opinion = 3
total_1a= muy_poco+suficiente+demasiado+sin_opinion

prop_muy_poco = muy_poco/total_1a
prop_suficiente = suficiente/total_1a
prop_demasiado = demasiado/total_1a
prop_sin_opinion = sin_opinion/total_1a


print(f"Proporción de 'Muy poco' = {prop_muy_poco} \nPorcentaje de 'Muy poco' = {prop_muy_poco*100}%\n{'-'*35}")
print(f"Proporción de 'Suficiente' = {prop_suficiente}\nPorcentaje de 'Suficiente' = {prop_suficiente*100}%\n{'-'*35}")
print(f"Proporción de 'Demasiado' = {prop_demasiado}\nPorcentaje de 'Demasiado' = {prop_demasiado*100}%\n{'-'*35}")
print(f"Proporción de 'Sin opinión' = {prop_sin_opinion}\nPorcentaje de 'Sin opinión' = {prop_sin_opinion*100}%\n{'-'*35}")


Proporción de 'Muy poco' = 0.772 
Porcentaje de 'Muy poco' = 77.2%
-----------------------------------
Proporción de 'Suficiente' = 0.206
Porcentaje de 'Suficiente' = 20.599999999999998%
-----------------------------------
Proporción de 'Demasiado' = 0.016
Porcentaje de 'Demasiado' = 1.6%
-----------------------------------
Proporción de 'Sin opinión' = 0.006
Porcentaje de 'Sin opinión' = 0.6%
-----------------------------------


### Parte B

In [3]:
muy_poco = 357
suficiente= 121
demasiado= 14
sin_opinion = 8
total_1a= muy_poco+suficiente+demasiado+sin_opinion

prop_muy_poco = muy_poco/total_1a
prop_suficiente = suficiente/total_1a
prop_demasiado = demasiado/total_1a
prop_sin_opinion = sin_opinion/total_1a


print(f"Proporción de 'Muy poco' = {prop_muy_poco} \nPorcentaje de 'Muy poco' = {prop_muy_poco*100}%\n{'-'*35}")
print(f"Proporción de 'Suficiente' = {prop_suficiente}\nPorcentaje de 'Suficiente' = {prop_suficiente*100}%\n{'-'*35}")
print(f"Proporción de 'Demasiado' = {prop_demasiado}\nPorcentaje de 'Demasiado' = {prop_demasiado*100}%\n{'-'*35}")
print(f"Proporción de 'Sin opinión' = {prop_sin_opinion}\nPorcentaje de 'Sin opinión' = {prop_sin_opinion*100}%\n{'-'*35}")

Proporción de 'Muy poco' = 0.714 
Porcentaje de 'Muy poco' = 71.39999999999999%
-----------------------------------
Proporción de 'Suficiente' = 0.242
Porcentaje de 'Suficiente' = 24.2%
-----------------------------------
Proporción de 'Demasiado' = 0.028
Porcentaje de 'Demasiado' = 2.8000000000000003%
-----------------------------------
Proporción de 'Sin opinión' = 0.016
Porcentaje de 'Sin opinión' = 1.6%
-----------------------------------


### Parte C

In [4]:
muy_poco = 413
suficiente= 67
demasiado= 10
sin_opinion = 10
total_1a= muy_poco+suficiente+demasiado+sin_opinion

prop_muy_poco = muy_poco/total_1a
prop_suficiente = suficiente/total_1a
prop_demasiado = demasiado/total_1a
prop_sin_opinion = sin_opinion/total_1a


print(f"Proporción de 'Muy poco' = {prop_muy_poco} \nPorcentaje de 'Muy poco' = {prop_muy_poco*100}%\n{'-'*35}")
print(f"Proporción de 'Suficiente' = {prop_suficiente}\nPorcentaje de 'Suficiente' = {prop_suficiente*100}%\n{'-'*35}")
print(f"Proporción de 'Demasiado' = {prop_demasiado}\nPorcentaje de 'Demasiado' = {prop_demasiado*100}%\n{'-'*35}")
print(f"Proporción de 'Sin opinión' = {prop_sin_opinion}\nPorcentaje de 'Sin opinión' = {prop_sin_opinion*100}%\n{'-'*35}")

Proporción de 'Muy poco' = 0.826 
Porcentaje de 'Muy poco' = 82.6%
-----------------------------------
Proporción de 'Suficiente' = 0.134
Porcentaje de 'Suficiente' = 13.4%
-----------------------------------
Proporción de 'Demasiado' = 0.02
Porcentaje de 'Demasiado' = 2.0%
-----------------------------------
Proporción de 'Sin opinión' = 0.02
Porcentaje de 'Sin opinión' = 2.0%
-----------------------------------


### Parte D

In [5]:
muy_poco = 126
suficiente= 142
demasiado= 215
sin_opinion = 17
total_1a= muy_poco+suficiente+demasiado+sin_opinion

prop_muy_poco = muy_poco/total_1a
prop_suficiente = suficiente/total_1a
prop_demasiado = demasiado/total_1a
prop_sin_opinion = sin_opinion/total_1a


print(f"Proporción de 'Muy poco' = {prop_muy_poco} \nPorcentaje de 'Muy poco' = {prop_muy_poco*100}%\n{'-'*35}")
print(f"Proporción de 'Suficiente' = {prop_suficiente}\nPorcentaje de 'Suficiente' = {prop_suficiente*100}%\n{'-'*35}")
print(f"Proporción de 'Demasiado' = {prop_demasiado}\nPorcentaje de 'Demasiado' = {prop_demasiado*100}%\n{'-'*35}")
print(f"Proporción de 'Sin opinión' = {prop_sin_opinion}\nPorcentaje de 'Sin opinión' = {prop_sin_opinion*100}%\n{'-'*35}")

Proporción de 'Muy poco' = 0.252 
Porcentaje de 'Muy poco' = 25.2%
-----------------------------------
Proporción de 'Suficiente' = 0.284
Porcentaje de 'Suficiente' = 28.4%
-----------------------------------
Proporción de 'Demasiado' = 0.43
Porcentaje de 'Demasiado' = 43.0%
-----------------------------------
Proporción de 'Sin opinión' = 0.034
Porcentaje de 'Sin opinión' = 3.4000000000000004%
-----------------------------------


## Ejercicio 2

In [6]:
b = 10
c= 4
a= 19-b-c
print(a)

5


## Ejercicio 3

In [7]:
n = 4
20 + n - 2

22

## Ejercicio 4

In [8]:
n = 20
media = 60
suma_x = media * n
print(suma_x)

1200


## Ejercicio 5

In [9]:
suma_x = 360
media = 90
n = suma_x / media
print(n)

4.0


## Ejercicio 6

In [10]:
sum_q=640
varianza= 16
N = sum_q/varianza
print(N)

40.0


## Ejercicio 7

In [11]:
suma_cuadrados = 240
n = 12
suma_cuadrados/n

20.0

## Ejercicio 8

Comparemos las dos funciones de redondeo más conocidas.

In [12]:
a= np.round(99.99500, 2)
print(a)
a= round(99.99500 , 2)
print(a)

print("-"*10)

b= np.round(46.40501, 2)
print(b)
b= round(46.40501, 2)
print(b)

print("-"*10)

c= np.round(2.96500, 2)
print(c)
c= round(2.96500, 2)
print(c)

print("-"*10)

d= np.round(0.00501, 2)
print(d)
d= round(0.00501, 2)
print(d)

print("-"*10)

e= np.round(16.46500, 2)
print(e)
e= round(16.46500, 2)
print(e)

print("-"*10)

f= np.round(1.05499, 2)
print(f)
f= round(1.05499, 2)
print(f)

print("-"*10)

g= np.round(86.2139, 2)
print(g)
g= round(86.2139, 2)
print(g)

print("-"*10)

h= np.round(10.0050, 2)
print(h)
h= round(10.0050, 2)
print(h)

100.0
100.0
----------
46.41
46.41
----------
2.96
2.96
----------
0.01
0.01
----------
16.46
16.46
----------
1.05
1.05
----------
86.21
86.21
----------
10.01
10.01


## Ejercicio 9

In [13]:
suicidios = [15840, 4625]
p_hombres = round(suicidios[0]/sum(suicidios),2)
p_mujeres = round(suicidios[1]/sum(suicidios),2)
print('Proporción de suicidios de hombres', p_hombres)
print('Proporción de suicidios de mujeres', p_mujeres)

Proporción de suicidios de hombres 0.77
Proporción de suicidios de mujeres 0.23


## Ejercicio 10

In [14]:
total = 17123
armas_fuego = round(11249/total,2)
arma_blanca = round(2895/total,2)
estran_golpes = round(1445/total,2)
objtos_contundentes = round(848/total,2)
incendio_provocado = round(173/total,2)
otros = round(423/total,2)

print('Proporción armas de fuego:', armas_fuego)
print('Proporción arma blanca:', arma_blanca)
print('Proporción estrangulamiento o golpes:', estran_golpes)
print('Proporción objetos contundentes:', objtos_contundentes)
print('Proporción incencdio provocado:', incendio_provocado)
print('Proporción otros:', otros)


Proporción armas de fuego: 0.66
Proporción arma blanca: 0.17
Proporción estrangulamiento o golpes: 0.08
Proporción objetos contundentes: 0.05
Proporción incencdio provocado: 0.01
Proporción otros: 0.02


## Ejercicio 11

In [15]:
datos_ej11 = [20465, 17123]
suicidios = round(datos_ej11[0]/sum(datos_ej11),4)
homicidios = round(datos_ej11[1]/sum(datos_ej11),4)
print('Proporción de suicidios:', suicidios)
print('Proporción de homicidios:', homicidios)

Proporción de suicidios: 0.5445
Proporción de homicidios: 0.4555


## Ejercicio 13

In [16]:
datos_ej13 = [4, 5, 7, 9, 10, 11, 14]

suma_a = sum(datos_ej13[0:4])
print('a)', suma_a, "\n", '-'*5)

suma_b = sum(datos_ej13[0:7])
print('b)', suma_b, "\n", '-'*5)

suma_c = sum(datos_ej13[2:6])
print('c)', suma_c, "\n", '-'*5)

suma_d = sum(datos_ej13[1:6])
print('d)', suma_d, "\n", '-'*5)

suma_e = sum(datos_ej13)
print('e)', suma_e, "\n", '-'*5)

suma_f = sum(datos_ej13[3::])
print('f)', suma_f)

a) 25 
 -----
b) 60 
 -----
c) 37 
 -----
d) 42 
 -----
e) 60 
 -----
f) 44


## Ejercicio 15

a) De razón.

b) De razón.

c) Nominal.

d) Ordinal.

## Ejercicio 17

In [17]:
a = round(np.sqrt(160), 2)
b = round(np.sqrt(16), 2)
c = round(np.sqrt(1.60), 2)
d = round(np.sqrt(0.16), 2)
e = round(np.sqrt(0.016), 2)

print([a,b,c,d,e])

[12.65, 4.0, 1.26, 0.4, 0.13]


## Ejercicio 18

a) (-0.5, 0.5) 

b) (0.45, 0.55)

c) (0.95, 1.05) *Esto es porque el dato está presentado con un decimal, de lo que se deduce que los datos tienen esa precisión en la medición. Por lo tanto, en lugar de operar con décimas, lo hacemos con centésimas.*

d) (0.485, 0.495)

e) (-5.5, -4.5)

f) (-4.55, -4.45)

## Ejercicio 19

In [18]:
suma_x2 = sum([x*x for x in datos_ej13])
print(f"Suma de cada dato al cuadrado: {suma_x2}")

cuadrado_suma_x = (sum(datos_ej13))**2
print(f"Cuadrado de la suma de cada dato: {cuadrado_suma_x}")

Suma de cada dato al cuadrado: 588
Cuadrado de la suma de cada dato: 3600


## Ejercicio 20

a) Discreta.

b) Discreta.

c) Continua.

d) Continua.

e) Continua.

f) Discreta.

## Ejercicios 21

In [19]:
import pandas as pd

datos_ej21 = pd.DataFrame({
    'Área académica' : ['Administración de empresas', 'Enseñanza', 'Humanidades', 'Ciencias', 'Ciencias sociales'],
    'Hombres': [400, 50, 150, 250, 200],
    'Mujeres': [100, 150, 200, 100, 200]
})

datos_ej21

Área académica  Hombres  Mujeres
0  Administración de empresas      400      100
1                   Enseñanza       50      150
2                 Humanidades      150      200
3                    Ciencias      250      100
4           Ciencias sociales      200      200

Para hacer los cáculos más fácilmente utilizando las herramientas de Pandas, agregamos una nueva columna a la matriz anterior en la que calculamos los totales por filas.

Usamos el método *sum()*, cuyo argumento *axis = 1* permite sumar por filas

In [20]:
datos_ej21['Total'] = datos_ej21[['Hombres', 'Mujeres']].sum(axis = 1)
datos_ej21

Área académica  Hombres  Mujeres  Total
0  Administración de empresas      400      100    500
1                   Enseñanza       50      150    200
2                 Humanidades      150      200    350
3                    Ciencias      250      100    350
4           Ciencias sociales      200      200    400

In [21]:
mujeres_xaa = [round(x*100,2) for x in (datos_ej21['Mujeres']/datos_ej21['Total'])]

datos_ej21['% Mujeres'] = mujeres_xaa # Lo agregamos al df para una mejor presentación de resultados
datos_ej21

Área académica  Hombres  Mujeres  Total  % Mujeres
0  Administración de empresas      400      100    500      20.00
1                   Enseñanza       50      150    200      75.00
2                 Humanidades      150      200    350      57.14
3                    Ciencias      250      100    350      28.57
4           Ciencias sociales      200      200    400      50.00

In [22]:
total_hombres = datos_ej21['Hombres'].sum()

p_hombres = [round((x/total_hombres)*100,2) for x in datos_ej21['Hombres']]

Esta vez, en lugar de añadir la lista al df, lo que hacemos es utilizar *zip()* para emparejar cada ítem del área académica con su respectivo porcentaje.

In [23]:
for n, x in zip(datos_ej21['Área académica'], p_hombres):
    print(f"{n}: {x}%")

Administración de empresas: 38.1%
Enseñanza: 4.76%
Humanidades: 14.29%
Ciencias: 23.81%
Ciencias sociales: 19.05%


In [24]:
total_mujeres = datos_ej21['Mujeres'].sum()

p_mujeres = [round((x/total_mujeres)*100,2) for x in datos_ej21['Mujeres']]
for n, x in zip(datos_ej21['Área académica'], p_mujeres):
    print(f"{n}: {x}%")

Administración de empresas: 13.33%
Enseñanza: 20.0%
Humanidades: 26.67%
Ciencias: 13.33%
Ciencias sociales: 26.67%


In [25]:
porcentaje_mujeres_aa = round((datos_ej21['Mujeres'].sum()/(datos_ej21['Mujeres'].sum() + datos_ej21['Hombres'].sum())*100), 2)
porcentaje_mujeres_aa


41.67

In [26]:
porcentaje_hombres_aa = 100 - porcentaje_mujeres_aa
porcentaje_hombres_aa

58.33

## Ejercicio 22

a) Continua.

b) Discreta.

c) Continua.

d) Discreta.

e) Discreta.

## Ejercicio 23

In [27]:
datos_ej23 = pd.DataFrame({
    "Año" : [1950, 1955, 1960, 1965, 1970],
    'Hombres' : [1824, 2074, 2180, 1927, 1915],
    'Mujeres': [1731, 1974, 2078, 1833, 1816]
})
datos_ej23

Año  Hombres  Mujeres
0  1950     1824     1731
1  1955     2074     1974
2  1960     2180     2078
3  1965     1927     1833
4  1970     1915     1816

In [28]:
datos_ej23['Total'] = datos_ej23[['Hombres', 'Mujeres']].sum(axis = 1)
datos_ej23['% Mujeres'] = round((datos_ej23['Mujeres']/datos_ej23['Total'])*100, 2)
datos_ej23['% Hombres'] = round((datos_ej23['Hombres']/datos_ej23['Total'])*100, 2)
datos_ej23

Año  Hombres  Mujeres  Total  % Mujeres  % Hombres
0  1950     1824     1731   3555      48.69      51.31
1  1955     2074     1974   4048      48.76      51.24
2  1960     2180     2078   4258      48.80      51.20
3  1965     1927     1833   3760      48.75      51.25
4  1970     1915     1816   3731      48.67      51.33

## Ejercicio 25

In [29]:
datos_ej24 = pd.DataFrame({
    'Medio de suicidio' : ['Armas de fuego', 'Envenenamiento', 'Asfixia', 'Ahorcamiento', 'Salto desde lugares elevados', 'Instrumentos cortantes'],
    'Berkeley' : [8, 6, 4, 2, 2, 1],
    'Yale': [10, 3, 5, 6, 1, 0]
})
datos_ej24

Medio de suicidio  Berkeley  Yale
0                Armas de fuego         8    10
1                Envenenamiento         6     3
2                       Asfixia         4     5
3                  Ahorcamiento         2     6
4  Salto desde lugares elevados         2     1
5        Instrumentos cortantes         1     0

In [30]:
p_berkeley = [round(x*100, 2) for x in datos_ej24['Berkeley']/datos_ej24['Berkeley'].sum()]
for n, x in zip(datos_ej24['Medio de suicidio'], p_berkeley):
    print(f"{n}: {x}%")

Armas de fuego: 34.78%
Envenenamiento: 26.09%
Asfixia: 17.39%
Ahorcamiento: 8.7%
Salto desde lugares elevados: 8.7%
Instrumentos cortantes: 4.35%


In [31]:
p_yale = [round(x*100, 2) for x in datos_ej24['Yale']/datos_ej24['Yale'].sum()]
for n, x in zip(datos_ej24['Medio de suicidio'], p_yale):
    print(f"{n}: {x}%")

Armas de fuego: 40.0%
Envenenamiento: 12.0%
Asfixia: 20.0%
Ahorcamiento: 24.0%
Salto desde lugares elevados: 4.0%
Instrumentos cortantes: 0.0%
